File: js/chineseNames.js

function nameGen(type){
	var nm1 = ["Ah","An","Bai","Bao","Bo","Cai","Chang","Chao","Chen","Cheng","Chin","Chun","Da","De","Delan","Dong","Fang","Fen","Feng","Fu","Gang","Gengxin","Guanting","Guanyu","Guiying","Guo","Hai","Hanying","He","Heng","Hong","Hu","Hua","Huan","Huang","Hui","Huo","Jia","Jiahao","Jian","Jiang","Jianhong","Jie","Jin","Jing","Jingyi","Ju","Jun","Kang","Kun","Lan","Lei","Li","Liang","Lim","Lin","Ling","Lingxin","Liuxian","Long","Luoyang","Meng","Min","Ming","Mu","Ning","Peng","Ping","Qiang","Qigang","Qing","Qiu","Rong","Ru","Ruogang","Shi","Shui","Shun","Shuren","Song","Su","Tai","Tao","Tian","Tu","Wei","Wen","Wu","Wuhan","Wuying","Xia","Xiang","Xieren","Xinya","Xinyi","Xinyue","Xiuying","Xue","Xuefeng","Xuegang","Xun","Yahui","Yan","Yang","Yating","Yazhu","Yi","Yijun","Yimu","Yin","Yong","Yu","Yuhan","Yun","Zan","Zedong","Zemin","Zexi","Zexian","Zhelan","Zhen","ZhenKang","Zheng","Zhenya","Zhi","Zhihao","Zhong","Zhou"];
	var nm2 = ["Ah","Ai","An","Bai","Bao","Bi","Bo","Cai","Chan","Chang","Chao","Chen","Cheng","Chin","Chun","Cui","Da","Dai","Dan","Delan","Fang","Fen","Feng","Fu","Gengxin","Guanting","Guanyu","Guiying","Guo","Hai","Hanying","He","Heng","Hong","Hua","Huan","Huang","Hui","Huo","Jia","Jian","Jiang","Jiao","Jie","Jin","Jing","Jingyi","Ju","Juan","Jun","Kun","Lan","Lei","Li","Lian","Lim","Lin","Ling","Lingxin","Liuxian","Luoyang","Mei","Min","Ming","My","Na","Ning","Nuan","Ping","Qigang","Qing","Qiu","Rong","Ru","Ruogang","Shi","Shu","Shufen","Shui","Shun","Shuren","Su","Tai","Tu","Wei","Wen","Wu","Wuhan","Wuying","Xia","Xiang","Xieren","Xinya","Xinyi","Xinyue","Xiu","Xiulan","Xiuying","Xuefeng","Xuegang","Xun","Ya","Yahui","Yaling","Yan","Yang","Yating","Yawen","Yazhu","Yi","Yijun","Yimu","Yong","Yu","Yuhan","Yun","Zan","Zedong","Zemin","Zexi","Zexian","Zhelan","Zhen","ZhenKang","Zheng","Zhenya","Zhi","Zhihao","Zhong","Zhou"];
	var nm3 = ["Bai","Cai","Cao","Chang","Chen","Cheng","Cui","Dai","Deng","Ding","Dong","Du","Duan","Fan","Fang","Feng","Fu","Gao","Geng","Gong","Gu","Guo","Han","Hao","He","Hou","Hu","Huang","Huo","Jia","Jiang","Jin","Kang","Kong","Lai","Lang","Lei","Li","Lian","Liang","Liao","Lin","Liu","Long","Lu","Luo","Ma","Mao","Meng","Mo","Pan","Peng","Qian","Qiao","Qin","Qiu","Quan","Ren","Shao","Shen","Shi","Song","Su","Sun","Tan","Tang","Tao","Teng","Tian","Wan","Wang","Wei","Wen","Wu","Xia","Xian","Xiang","Xiao","Xie","Xing","Xiong","Xu","Xuan","Xue","Xun","Yan","Yang","Yao","Ye","Yi","Yin","Yu","Yuan","Zeng","Zhan","Zhang","Zhao","Zhen","Zheng","Zhong","Zhou","Zhu","Zhuan","Zi","Zou"];
	var nm4 = ["Ah","An","Bai","Bao","Bo","Cai","Chang","Chao","Chen","Cheng","Chin","Chun","Da","De","Delan","Fang","Fen","Feng","Fu","Gengxin","Guanting","Guanyu","Guiying","Guo","Hai","Hanying","He","Heng","Hong","Hu","Hua","Huan","Huang","Hui","Huo","Jia","Jian","Jiang","Jie","Jin","Jing","Jingyi","Ju","Jun","Kang","Kun","Lan","Lei","Li","Liang","Lim","Lin","Ling","Lingxin","Liuxian","Long","Luoyang","Meng","Min","Ming","Mu","Ning","Peng","Ping","Qigang","Qing","Qiu","Rong","Ru","Ruogang","Shi","Shui","Shun","Shuren","Song","Su","Tai","Tian","Tu","Wei","Wen","Wu","Wuhan","Wuying","Xia","Xiang","Xieren","Xinya","Xinyi","Xinyue","Xiuying","Xue","Xuefeng","Xuegang","Xun","Yahui","Yan","Yang","Yating","Yazhu","Yi","Yijun","Yimu","Yin","Yong","Yu","Yuhan","Yun","Zan","Zedong","Zemin","Zexi","Zexian","Zhelan","Zhen","ZhenKang","Zheng","Zhenya","Zhi","Zhihao","Zhong","Zhou"];
	
	var br = "";

	$('#placeholder').css('textTransform', 'capitalize');
	var tp = type;
	var element = document.createElement("div");
	element.setAttribute("id", "result");
	
	for(i = 0; i < 10; i++){
		rnd2 = Math.random() * nm3.length | 0;
		if(tp === 1){
			rnd = Math.random() * nm2.length | 0;
			names = nm3[rnd2] + " " + nm2[rnd];
			nm2.splice(rnd, 1);
		}else if(tp === 2){
			rnd = Math.random() * nm4.length | 0;
			names = nm3[rnd2] + " " + nm4[rnd];
			nm4.splice(rnd, 1);
		}else{
			rnd = Math.random() * nm1.length | 0;
			names = nm3[rnd2] + " " + nm1[rnd];
			nm1.splice(rnd, 1);
		}
		nm3.splice(rnd2, 1);
		br = document.createElement('br');	
		element.appendChild(document.createTextNode(names));
		element.appendChild(br);
	}
	if(document.getElementById("result")){
		document.getElementById("placeholder").removeChild(document.getElementById("result"));
	}		
	document.getElementById("placeholder").appendChild(element);
}

File: generators/chineseNames.py
__all__ = ['chineseNames']

# Don't look below, you will not understand this Python code :) I don't.

from js2py.pyjs import *
# setting scope
var = Scope( JS_BUILTINS )
set_global_object(var)

# Code follows:
var.registers(['nameGen'])
@Js
def PyJsHoisted_nameGen_(type, this, arguments, var=var):
    var = Scope({'arguments':arguments, 'this':this, 'type':type}, var)
    var.registers(['nm1', 'nm4', 'type', 'nm3', 'tp', 'nm2', 'result'])
    var.put('nm1', Js([Js('Ah'), Js('An'), Js('Bai'), Js('Bao'), Js('Bo'), Js('Cai'), Js('Chang'), Js('Chao'), Js('Chen'), Js('Cheng'), Js('Chin'), Js('Chun'), Js('Da'), Js('De'), Js('Delan'), Js('Dong'), Js('Fang'), Js('Fen'), Js('Feng'), Js('Fu'), Js('Gang'), Js('Gengxin'), Js('Guanting'), Js('Guanyu'), Js('Guiying'), Js('Guo'), Js('Hai'), Js('Hanying'), Js('He'), Js('Heng'), Js('Hong'), Js('Hu'), Js('Hua'), Js('Huan'), Js('Huang'), Js('Hui'), Js('Huo'), Js('Jia'), Js('Jiahao'), Js('Jian'), Js('Jiang'), Js('Jianhong'), Js('Jie'), Js('Jin'), Js('Jing'), Js('Jingyi'), Js('Ju'), Js('Jun'), Js('Kang'), Js('Kun'), Js('Lan'), Js('Lei'), Js('Li'), Js('Liang'), Js('Lim'), Js('Lin'), Js('Ling'), Js('Lingxin'), Js('Liuxian'), Js('Long'), Js('Luoyang'), Js('Meng'), Js('Min'), Js('Ming'), Js('Mu'), Js('Ning'), Js('Peng'), Js('Ping'), Js('Qiang'), Js('Qigang'), Js('Qing'), Js('Qiu'), Js('Rong'), Js('Ru'), Js('Ruogang'), Js('Shi'), Js('Shui'), Js('Shun'), Js('Shuren'), Js('Song'), Js('Su'), Js('Tai'), Js('Tao'), Js('Tian'), Js('Tu'), Js('Wei'), Js('Wen'), Js('Wu'), Js('Wuhan'), Js('Wuying'), Js('Xia'), Js('Xiang'), Js('Xieren'), Js('Xinya'), Js('Xinyi'), Js('Xinyue'), Js('Xiuying'), Js('Xue'), Js('Xuefeng'), Js('Xuegang'), Js('Xun'), Js('Yahui'), Js('Yan'), Js('Yang'), Js('Yating'), Js('Yazhu'), Js('Yi'), Js('Yijun'), Js('Yimu'), Js('Yin'), Js('Yong'), Js('Yu'), Js('Yuhan'), Js('Yun'), Js('Zan'), Js('Zedong'), Js('Zemin'), Js('Zexi'), Js('Zexian'), Js('Zhelan'), Js('Zhen'), Js('ZhenKang'), Js('Zheng'), Js('Zhenya'), Js('Zhi'), Js('Zhihao'), Js('Zhong'), Js('Zhou')]))
    var.put('nm2', Js([Js('Ah'), Js('Ai'), Js('An'), Js('Bai'), Js('Bao'), Js('Bi'), Js('Bo'), Js('Cai'), Js('Chan'), Js('Chang'), Js('Chao'), Js('Chen'), Js('Cheng'), Js('Chin'), Js('Chun'), Js('Cui'), Js('Da'), Js('Dai'), Js('Dan'), Js('Delan'), Js('Fang'), Js('Fen'), Js('Feng'), Js('Fu'), Js('Gengxin'), Js('Guanting'), Js('Guanyu'), Js('Guiying'), Js('Guo'), Js('Hai'), Js('Hanying'), Js('He'), Js('Heng'), Js('Hong'), Js('Hua'), Js('Huan'), Js('Huang'), Js('Hui'), Js('Huo'), Js('Jia'), Js('Jian'), Js('Jiang'), Js('Jiao'), Js('Jie'), Js('Jin'), Js('Jing'), Js('Jingyi'), Js('Ju'), Js('Juan'), Js('Jun'), Js('Kun'), Js('Lan'), Js('Lei'), Js('Li'), Js('Lian'), Js('Lim'), Js('Lin'), Js('Ling'), Js('Lingxin'), Js('Liuxian'), Js('Luoyang'), Js('Mei'), Js('Min'), Js('Ming'), Js('My'), Js('Na'), Js('Ning'), Js('Nuan'), Js('Ping'), Js('Qigang'), Js('Qing'), Js('Qiu'), Js('Rong'), Js('Ru'), Js('Ruogang'), Js('Shi'), Js('Shu'), Js('Shufen'), Js('Shui'), Js('Shun'), Js('Shuren'), Js('Su'), Js('Tai'), Js('Tu'), Js('Wei'), Js('Wen'), Js('Wu'), Js('Wuhan'), Js('Wuying'), Js('Xia'), Js('Xiang'), Js('Xieren'), Js('Xinya'), Js('Xinyi'), Js('Xinyue'), Js('Xiu'), Js('Xiulan'), Js('Xiuying'), Js('Xuefeng'), Js('Xuegang'), Js('Xun'), Js('Ya'), Js('Yahui'), Js('Yaling'), Js('Yan'), Js('Yang'), Js('Yating'), Js('Yawen'), Js('Yazhu'), Js('Yi'), Js('Yijun'), Js('Yimu'), Js('Yong'), Js('Yu'), Js('Yuhan'), Js('Yun'), Js('Zan'), Js('Zedong'), Js('Zemin'), Js('Zexi'), Js('Zexian'), Js('Zhelan'), Js('Zhen'), Js('ZhenKang'), Js('Zheng'), Js('Zhenya'), Js('Zhi'), Js('Zhihao'), Js('Zhong'), Js('Zhou')]))
    var.put('nm3', Js([Js('Bai'), Js('Cai'), Js('Cao'), Js('Chang'), Js('Chen'), Js('Cheng'), Js('Cui'), Js('Dai'), Js('Deng'), Js('Ding'), Js('Dong'), Js('Du'), Js('Duan'), Js('Fan'), Js('Fang'), Js('Feng'), Js('Fu'), Js('Gao'), Js('Geng'), Js('Gong'), Js('Gu'), Js('Guo'), Js('Han'), Js('Hao'), Js('He'), Js('Hou'), Js('Hu'), Js('Huang'), Js('Huo'), Js('Jia'), Js('Jiang'), Js('Jin'), Js('Kang'), Js('Kong'), Js('Lai'), Js('Lang'), Js('Lei'), Js('Li'), Js('Lian'), Js('Liang'), Js('Liao'), Js('Lin'), Js('Liu'), Js('Long'), Js('Lu'), Js('Luo'), Js('Ma'), Js('Mao'), Js('Meng'), Js('Mo'), Js('Pan'), Js('Peng'), Js('Qian'), Js('Qiao'), Js('Qin'), Js('Qiu'), Js('Quan'), Js('Ren'), Js('Shao'), Js('Shen'), Js('Shi'), Js('Song'), Js('Su'), Js('Sun'), Js('Tan'), Js('Tang'), Js('Tao'), Js('Teng'), Js('Tian'), Js('Wan'), Js('Wang'), Js('Wei'), Js('Wen'), Js('Wu'), Js('Xia'), Js('Xian'), Js('Xiang'), Js('Xiao'), Js('Xie'), Js('Xing'), Js('Xiong'), Js('Xu'), Js('Xuan'), Js('Xue'), Js('Xun'), Js('Yan'), Js('Yang'), Js('Yao'), Js('Ye'), Js('Yi'), Js('Yin'), Js('Yu'), Js('Yuan'), Js('Zeng'), Js('Zhan'), Js('Zhang'), Js('Zhao'), Js('Zhen'), Js('Zheng'), Js('Zhong'), Js('Zhou'), Js('Zhu'), Js('Zhuan'), Js('Zi'), Js('Zou')]))
    var.put('nm4', Js([Js('Ah'), Js('An'), Js('Bai'), Js('Bao'), Js('Bo'), Js('Cai'), Js('Chang'), Js('Chao'), Js('Chen'), Js('Cheng'), Js('Chin'), Js('Chun'), Js('Da'), Js('De'), Js('Delan'), Js('Fang'), Js('Fen'), Js('Feng'), Js('Fu'), Js('Gengxin'), Js('Guanting'), Js('Guanyu'), Js('Guiying'), Js('Guo'), Js('Hai'), Js('Hanying'), Js('He'), Js('Heng'), Js('Hong'), Js('Hu'), Js('Hua'), Js('Huan'), Js('Huang'), Js('Hui'), Js('Huo'), Js('Jia'), Js('Jian'), Js('Jiang'), Js('Jie'), Js('Jin'), Js('Jing'), Js('Jingyi'), Js('Ju'), Js('Jun'), Js('Kang'), Js('Kun'), Js('Lan'), Js('Lei'), Js('Li'), Js('Liang'), Js('Lim'), Js('Lin'), Js('Ling'), Js('Lingxin'), Js('Liuxian'), Js('Long'), Js('Luoyang'), Js('Meng'), Js('Min'), Js('Ming'), Js('Mu'), Js('Ning'), Js('Peng'), Js('Ping'), Js('Qigang'), Js('Qing'), Js('Qiu'), Js('Rong'), Js('Ru'), Js('Ruogang'), Js('Shi'), Js('Shui'), Js('Shun'), Js('Shuren'), Js('Song'), Js('Su'), Js('Tai'), Js('Tian'), Js('Tu'), Js('Wei'), Js('Wen'), Js('Wu'), Js('Wuhan'), Js('Wuying'), Js('Xia'), Js('Xiang'), Js('Xieren'), Js('Xinya'), Js('Xinyi'), Js('Xinyue'), Js('Xiuying'), Js('Xue'), Js('Xuefeng'), Js('Xuegang'), Js('Xun'), Js('Yahui'), Js('Yan'), Js('Yang'), Js('Yating'), Js('Yazhu'), Js('Yi'), Js('Yijun'), Js('Yimu'), Js('Yin'), Js('Yong'), Js('Yu'), Js('Yuhan'), Js('Yun'), Js('Zan'), Js('Zedong'), Js('Zemin'), Js('Zexi'), Js('Zexian'), Js('Zhelan'), Js('Zhen'), Js('ZhenKang'), Js('Zheng'), Js('Zhenya'), Js('Zhi'), Js('Zhihao'), Js('Zhong'), Js('Zhou')]))
    var.put('tp', var.get('type'))
    var.put('result', Js([]))
    #for JS loop
    var.put('i', Js(0.0))
    while (var.get('i')<Js(10.0)):
        try:
            var.put('rnd2', ((var.get('Math').callprop('random')*var.get('nm3').get('length'))|Js(0.0)))
            if PyJsStrictEq(var.get('tp'),Js(1.0)):
                var.put('rnd', ((var.get('Math').callprop('random')*var.get('nm2').get('length'))|Js(0.0)))
                var.put('names', ((var.get('nm3').get(var.get('rnd2'))+Js(' '))+var.get('nm2').get(var.get('rnd'))))
                var.get('nm2').callprop('splice', var.get('rnd'), Js(1.0))
            else:
                if PyJsStrictEq(var.get('tp'),Js(2.0)):
                    var.put('rnd', ((var.get('Math').callprop('random')*var.get('nm4').get('length'))|Js(0.0)))
                    var.put('names', ((var.get('nm3').get(var.get('rnd2'))+Js(' '))+var.get('nm4').get(var.get('rnd'))))
                    var.get('nm4').callprop('splice', var.get('rnd'), Js(1.0))
                else:
                    var.put('rnd', ((var.get('Math').callprop('random')*var.get('nm1').get('length'))|Js(0.0)))
                    var.put('names', ((var.get('nm3').get(var.get('rnd2'))+Js(' '))+var.get('nm1').get(var.get('rnd'))))
                    var.get('nm1').callprop('splice', var.get('rnd'), Js(1.0))
            var.get('nm3').callprop('splice', var.get('rnd2'), Js(1.0))
            var.get('result').callprop('push', var.get('names'))
        finally:
                (var.put('i',Js(var.get('i').to_number())+Js(1))-Js(1))
    pass
    return var.get('result')
PyJsHoisted_nameGen_.func_name = 'nameGen'
var.put('nameGen', PyJsHoisted_nameGen_)
pass
pass


# Add lib to the module scope
chineseNames = var.to_python()